pkg/webhook/pod/webhook.go

```diff
@@ -57,44 +59,118 @@ type WebhookServer struct {
 }
 
 func (wh *WebhookServer) Handle(ctx context.Context, req admission.Request) admission.Response {
-	pod := &corev1.Pod{}
-	err := wh.decoder.Decode(req, pod)
-	if err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
-	log := logger.FromContext(ctx)
 
-	// handle empty namespace field when the pod is created by deployment
-	if pod.ObjectMeta.Namespace == "" {
-		pod.ObjectMeta.Namespace = req.Namespace
-	}
+	log.Info("logging the req obj", "req", req)
 
-	if mutate, sliceName := wh.MutationRequired(pod.ObjectMeta, ctx); !mutate {
-		log.Info("mutation not required", "pod metadata", pod.ObjectMeta)
-	} else {
-		log.Info("mutating pod", "pod metadata", pod.ObjectMeta)
-		pod = Mutate(pod, sliceName)
-		log.Info("mutated pod", "pod metadata", pod.ObjectMeta)
-	}
+	if req.Kind.Kind == "Pod" {
+		pod := &corev1.Pod{}
+		err := wh.decoder.Decode(req, pod)
+		if err != nil {
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+		log := logger.FromContext(ctx)
 
-	marshaled, err := json.Marshal(pod)
-	if err != nil {
-		return admission.Errored(http.StatusInternalServerError, err)
+		// handle empty namespace field when the pod is created by deployment
+		if pod.ObjectMeta.Namespace == "" {
+			pod.ObjectMeta.Namespace = req.Namespace
+		}
+
+		if mutate, sliceName := wh.MutationRequired(pod.ObjectMeta, ctx, req.Kind.Kind); !mutate {
+			log.Info("mutation not required", "pod metadata", pod.ObjectMeta)
+		} else {
+			log.Info("mutating pod", "pod metadata", pod.ObjectMeta)
+			pod = MutatePod(pod, sliceName)
+			log.Info("mutated pod", "pod metadata", pod.ObjectMeta)
+		}
+
+		marshaled, err := json.Marshal(pod)
+		if err != nil {
+			return admission.Errored(http.StatusInternalServerError, err)
+		}
+		return admission.PatchResponseFromRaw(req.Object.Raw, marshaled)
+	} else if req.Kind.Kind == "Deployment" {
+		deploy := &appsv1.Deployment{}
+		log := logger.FromContext(ctx)
+		err := wh.decoder.Decode(req, deploy)
+		if err != nil {
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+
+		if mutate, sliceName := wh.MutationRequired(deploy.ObjectMeta, ctx, req.Kind.Kind); !mutate {
+			log.Info("mutation not required", "pod metadata", deploy.Spec.Template.ObjectMeta)
```

Comment:
please correct the log - "pod metadata" to "deploy metadata"

Reply:
It is the pod metadata present inside deployment